engine.go

```diff
@@ -635,19 +637,15 @@ func (e *Engine) WithBackgroundThreads(b *sql.BackgroundThreads) *Engine {
 	return e
 }
 
+func (e *Engine) IsReadOnly() bool {
+	return e.ReadOnly.Load()
+}
+
 // readOnlyCheck checks to see if the query is valid with the modification setting of the engine.
 func (e *Engine) readOnlyCheck(node sql.Node) error {
-	if plan.IsDDLNode(node) {
-		if e.IsReadOnly {
-			return sql.ErrReadOnly.New()
-		} else if e.IsServerLocked {
-			return sql.ErrDatabaseWriteLocked.New()
-		}
-	}
-	switch node.(type) {
-	case
-		*plan.DeleteFrom, *plan.InsertInto, *plan.Update, *plan.LockTables, *plan.UnlockTables:
-		if e.IsReadOnly {
+	nodeIsReadOnly := plan.IsReadOnly(node)
+	if !nodeIsReadOnly {
+		if e.IsReadOnly() {
```

Comment:
would it be incorrect to invert this check? I'd think e.IsReadyOnly() is less expensive

Reply:
Great point. Just pushed a fix. PTAL.